core/app/models/spree/ability.rb

```diff
@@ -57,6 +57,8 @@ def alias_cancan_delete_action
 
     def apply_admin_permissions(user)
       can :manage, :all
+      cannot :manage, ::Spree::CreditCard
+      can :manage, ::Spree::CreditCard, user_id: user.id
```

Comment:
@szymoniwacz no, this needs to stay as it was before, we only need to modify the scope in the API v2 controller